### PR TITLE
feat(docs): act-1481 - improvements to reference pages

### PR DESCRIPTION
**Pull request by TrofimovAnton85**
## Description

- added `extensionOnly` support to mm SDK
- added condition for initialization
- added `test-id` attribute for Sentry statistics

https://infura.atlassian.net/browse/ACT-1481

### CODE REVIEW

**Review by Voltmod — DISMISSED** [review]

**Review by aednikanov — DISMISSED** [review]

File: src/theme/Root.tsx [hunk 1]
```diff
@@ -31,6 +32,12 @@ export default function Root({ children }: { children: ReactChild}) {
     }
   }
```
Comment (joaniefromtheblock) [inline src/theme/Root.tsx:34]: Do we need to initialize the  state for `metaMaskProvider`?

File: src/theme/Root.tsx [hunk 2]
```diff
@@ -31,6 +32,12 @@ export default function Root({ children }: { children: ReactChild}) {
...
 
+  useEffect(() => {
+    if (sdk && !metaMaskProvider) {
+      metaMaskConnectHandler();
+    }
+  }, [metaMaskProvider]);
+
   return (
```
Comment (joaniefromtheblock) [inline src/theme/Root.tsx:41]:
```suggestion
return (
  <MetamaskProviderContext.Provider value={metaMaskProvider}>
    {children}
  </MetamaskProviderContext.Provider>
);
```

File: src/theme/Root.tsx [hunk 3]
```diff
@@ -31,6 +32,12 @@ export default function Root({ children }: { children: ReactChild}) {
...
     <MetamaskProviderContext.Provider value={{
       metaMaskProvider,
```
Comment (joaniefromtheblock) [inline src/theme/Root.tsx:43]: Is this complete?

**Review by joaniefromtheblock — DISMISSED** [review]
LGTM, but have a couple of questions

**Comment by TrofimovAnton85** [comment]
> LGTM, but have a couple of questions

Yes, it's completed. Sorry if there was confusion.
I corrected the initialization of the provider - now if the user has the extension installed, then we have verification and automatic connection.
If the extension is not installed, then the modal (with install/connect to wallet) will be as usual by the "connect" button.

please, re-approve.

**Review by joaniefromtheblock — APPROVED** [review]
LGTM. Thanks a million Anton

**Review by aednikanov — APPROVED** [review]

**Pull request merged**